frontend/src/features/documents/documentDetail/DocumentDetailView.test.tsx

```diff
@@ -0,0 +1,116 @@
+import { Claims } from 'constants/claims';
+import { createMemoryHistory } from 'history';
+import { mockLookups } from 'mocks/mockLookups';
+import { Api_DocumentType } from 'models/api/Document';
+import { Api_Storage_DocumentMetadata } from 'models/api/DocumentStorage';
+import { lookupCodesSlice } from 'store/slices/lookupCodes';
+import { mockKeycloak, render, RenderOptions } from 'utils/test-utils';
+
+import { ComposedDocument } from '../ComposedDocument';
+import { DocumentDetailView } from './DocumentDetailView';
+
+// mock auth library
+jest.mock('@react-keycloak/web');
+
+const history = createMemoryHistory();
+
+const documentTypes: Api_DocumentType[] = [
+  {
+    id: 1,
+    documentType: 'BC Assessment Search',
+    mayanId: 17,
+  },
+  {
+    id: 2,
+    documentType: 'Privy Council',
+    mayanId: 7,
+  },
+];
+const documentTypeMetadata: Api_Storage_DocumentMetadata[] = [
+  {
+    document: {
+      label: '',
+      datetime_created: '2022-07-27T16:06:42.42',
+      description: '',
+      file_latest: {
+        id: 2,
+        document_id: 1,
+        comment: '',
+        encoding: '',
+        fileName: '',
+        mimetype: '',
+        size: 12,
+        timeStamp: '',
+      },
+      id: 1,
+      document_type: { id: 1, label: 'BC Assessment Search' },
+    },
+    id: 1,
+    metadata_type: { id: 1, label: 'Tag' },
+    url: '',
+    value: 'Tag1234',
+  },
+];
+const mockDocument: ComposedDocument = {
+  mayanMetadata: documentTypeMetadata,
+  pimsDocument: {
+    id: 1,
+    mayanDocumentId: 15,
+    documentType: documentTypes[0],
+    statusTypeCode: { id: 'AMEND', description: 'Amended' },
+    fileName: 'NewFile.doc',
+  },
+};
+describe('DocumentDetailView component', () => {
+  // render component under test
+  const setup = (renderOptions: RenderOptions) => {
+    const utils = render(
+      <DocumentDetailView onUpdate={jest.fn()} document={mockDocument} isLoading={false} />,
+      {
+        ...renderOptions,
+        store: {
+          [lookupCodesSlice.name]: { lookupCodes: mockLookups },
+        },
+        history,
+      },
+    );
+
+    return {
+      ...utils,
+      useMockAuthentication: true,
+    };
+  };
+
+  beforeEach(() => {
+    mockKeycloak({ claims: [Claims.DOCUMENT_VIEW, Claims.DOCUMENT_EDIT] });
```

Comment:
for simplicity you could pass the claims to the render method in the setup function above

Reply:
done